src/AppInstallerCLITests/RestInterface_1_0.cpp

```diff
@@ -0,0 +1,413 @@
+    REQUIRE(package.Versions.at(1).VersionAndChannel.GetVersion().ToString().compare("2.0.0") == 0);
+}
+
+TEST_CASE("Search_BadResponse", "[RestSource]")
+{
+    utility::string_t sample = _XPLATSTR(
+        R"delimeter({
+            "Data" : [
+               {
+              "PackageIdentifier": "git.package",
+              "PackageName": "package",
+              "Publisher": "git",
+              "Versions": null }]
+        })delimeter");
+
+    HttpClientHelper helper{ GetTestHandler(web::http::status_codes::OK, std::move(sample)) };
+    Interface v1{ TestRestUriString, std::move(helper) };
+    REQUIRE_THROWS_HR(v1.Search({}), APPINSTALLER_CLI_ERROR_RESTSOURCE_INVALID_DATA);
+}
+
+TEST_CASE("Search_BadResponse_NotFoundCode", "[RestSource]")
+{
+    HttpClientHelper helper{ GetTestHandler(web::http::status_codes::NotFound) };
+    Interface v1{ TestRestUriString, std::move(helper) };
+    Schema::IRestClient::SearchResult result = v1.Search({});
+    REQUIRE(result.Matches.empty());
+}
+
+TEST_CASE("Search_Optimized_ManifestResponse", "[RestSource]")
+{
+    utility::string_t sample = GetGoodManifest_RequiredFields();
+    HttpClientHelper helper{ GetTestHandler(web::http::status_codes::OK, std::move(sample)) };
+    AppInstaller::Repository::SearchRequest request;
+    PackageMatchFilter filter{ PackageMatchField::Id, MatchType::Exact, "Foo" };
+    request.Filters.emplace_back(std::move(filter));
+    Interface v1{ TestRestUriString, std::move(helper) };
+    Schema::IRestClient::SearchResult result = v1.Search(request);
+    REQUIRE(result.Matches.size() == 1);
+    REQUIRE(result.Matches[0].Versions.size() == 1);
+    REQUIRE(result.Matches[0].Versions[0].VersionAndChannel.GetVersion().ToString() == "5.0.0");
+    REQUIRE(result.Matches[0].Versions[0].VersionAndChannel.GetChannel().ToString() == "");
+    REQUIRE(result.Matches[0].Versions[0].Manifest);
+    
+    // Verify manifest is populated
+    Manifest manifest = result.Matches[0].Versions[0].Manifest.value();
+    REQUIRE(manifest.Id == "Foo.Bar");
+    REQUIRE(manifest.Version == "5.0.0");
+    REQUIRE(manifest.DefaultLocalization.Locale == "en-us");
+    REQUIRE(manifest.DefaultLocalization.Get<Localization::Publisher>() == "Foo");
+    REQUIRE(manifest.DefaultLocalization.Get<Localization::PackageName>() == "Bar");
+    REQUIRE(manifest.DefaultLocalization.Get<Localization::License>() == "Foo bar license");
+    REQUIRE(manifest.DefaultLocalization.Get<Localization::ShortDescription>() == "Foo bar description");
+    REQUIRE(manifest.Installers.size() == 1);
+    REQUIRE(manifest.Installers[0].Arch == Architecture::X64);
+    REQUIRE(manifest.Installers[0].Sha256 == AppInstaller::Utility::SHA256::ConvertToBytes("011048877dfaef109801b3f3ab2b60afc74f3fc4f7b3430e0c897f5da1df84b6"));
+    REQUIRE(manifest.Installers[0].InstallerType == InstallerTypeEnum::Exe);
+    REQUIRE(manifest.Installers[0].Url == "https://dummyurl.com/foobar.exe");

```

Comment:
Unless you really need to, you should be able to use something like:
```suggestion
    REQUIRE(manifest.Installers[0].Url == "https://installer.example.com/foobar.exe");
```